library/core/src/iter/sources/repeat.rs

```diff
@@ -8,11 +8,15 @@ use crate::num::NonZero;
 /// Infinite iterators like `repeat()` are often used with adapters like
 /// [`Iterator::take()`], in order to make them finite.
 ///
+/// > Please prefer [`str::repeat()`] over this function if you just want
+/// > to repeat a char/string `n`th times.
```

Comment:
Why the block quote?  I'd expect this to be phrased as more of an optimization hint, similar to how the guidance on https://doc.rust-lang.org/nightly/std/collections/index.html doesn't really say "please" and instead focuses on providing information.